mycroft/skills/skill_loader.py

```diff
@@ -355,3 +357,64 @@ def _communicate_load_status(self):
             )
             self.bus.emit(message)
             LOG.error('Skill {} failed to load'.format(self.skill_id))
+
+
+class PluginSkillLoader(SkillLoader):
+    def __init__(self, bus, skill_id):
+        super().__init__(bus, skill_id)
+        self.skill_directory = skill_id
+        self.skill_id = skill_id
+
+    def reload_needed(self):
+        return False
+
+    def _create_skill_instance(self, skill_module):
+        """Use v2 skills framework to create the skill."""
+        try:
+            self.instance = skill_module()
+        except Exception as e:
+            log_msg = 'Skill __init__ failed with {}'
+            LOG.exception(log_msg.format(repr(e)))
+            self.instance = None
+
+        if self.instance:
+            self.instance.skill_id = self.skill_id
```

Comment:
#11 is needed, otherwise init_settings is only called before skill_id is set for plugin skills